cmds/group/actors.go

```diff
@@ -103,3 +117,48 @@ func runCancel(credentials *tcclient.Credentials, args []string, out io.Writer,
 		return nil
 	}
 }
+
+// filterTask takes a task and returns whether or not this task should be
+// set for cancellation, based on the specified filters through flags
+func filterTask(status queue.TaskStatusStructure, flags *pflag.FlagSet) bool {
+	// first check - only delete tasks that are unscheduled, pending, running
+	if status.State != "unscheduled" && status.State != "pending" && status.State != "running" {
+		return false
+	}
+
+	// filter for worker type, if one specified
+	// if no worker type is specified, its value is "" so the condition is skipped
+	if workerType, _ := flags.GetString("worker-type"); workerType != "" {
+		if workerType != status.WorkerType {
+			return false
+		}
+	}
+
+	// TODO add other filters here when necessary
+
+	return true
+}
+
+// confirmCancellation lists the tasks to be cancelled and prompts to confirm cancellation
+func confirmCancellation(tasks []string, out io.Writer) bool {
+	// list tasks
+	fmt.Fprintf(out, "The following %d tasks will be cancelled:\n", len(tasks))
+
+	for _, task := range tasks {
+		fmt.Fprintf(out, "\tTask %s\n", task)
+	}
+
+	for {
+		fmt.Fprint(out, "Are you sure you want to cancel these tasks? [Y/n] ")
```

Comment:
I like the fact "yes" is not the default choice 😃 

Nit: I think the choice in upper case is generally the default choice. I'd put "Y" in lower case, then